### PR TITLE
Grant/bru-1060

### PR DESCRIPTION
**Pull request by glpierce**
Implement relate step & take step for query builder

### CODE REVIEW

File: backend/src/user-queries/query-parsing/parse-relate.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import * as _ from "lodash";
+import * as assert from "assert";
+
+const relateTemplate = `
+let {{ stepName }} = (
+  from {{ from }}
+  join side:left {{ relation }} (this.{{ baseJoinColumn }}=={{ relation }}.{{ relationJoinColumn }})
+)`;
+
+const relateTemplateManyToMany = `
```
Comment (tevonsb) [inline backend/src/user-queries/query-parsing/parse-relate.ts:18]: These are really pretty, nicely done

File: frontend/src/components/column-selectors/single-column-selector/single-column-selector-list-item.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          <Text className="ml-1">{`(${column.relation.as})`}</Text>
+        )}
+      </div>
+    );
+  }
+
+  return (
+    <MenuItem
+      onClick={handleClick}
+      selected={_.isEqual(selected, column)}
```
Comment (tevonsb) [inline frontend/src/components/column-selectors/single-column-selector/single-column-selector-list-item.tsx:46]: Hmm, do we need the deep equality? can't we just check the names? we should have at least some piece of unique data here

Comment (tevonsb) [reply frontend/src/components/column-selectors/single-column-selector/single-column-selector-list-item.tsx:46]: Deep equals is just a more expensive operation

File: frontend/src/components/column-selectors/single-column-selector/single-column-selector.tsx [hunk 3]
```diff
@@ -40,9 +66,11 @@ export default function SingleColumnSelector({
       itemRenderer={renderColumnItem}
       onItemSelect={onColumnSelect}
     >
-      <Button rightIcon="double-caret-vertical" disabled={disabled === true}>
-        {selected ? selected.name : "Select a column"}
-      </Button>
+      <SingleColumnSelectorButton
+        selected={selected}
+        disabled={!!disabled}
```
Comment (tevonsb) [inline frontend/src/components/column-selectors/single-column-selector/single-column-selector.tsx:71]: Why do we need to do this? Is it not already set? Likely indicates that we should refactor if "disabled" needs to be cast to a boolean and isn't already

File: frontend/src/components/query/query-builder/query-builder.tsx [hunk 4]
```diff
@@ -115,10 +119,35 @@ export default function QueryBuilder({
           />
         );
       case StepIdentifierEnum.Relate:
+        return (
+          <RelateStepComponent
+            key={create ? "new step" : index}
+            index={index}
+            step={!!step ? (step as RelateStep) : step}
```
Comment (tevonsb) [inline frontend/src/components/query/query-builder/query-builder.tsx:126]: Do we need !!? step will evaluate as a condition without the conversion to boolean

**Review by tevonsb — APPROVED** [review]
Approved with some small changes! Feel free to make in separate branch if want to get this merged, not functional differences

File: frontend/src/components/column-selectors/single-column-selector/single-column-selector-list-item.tsx (hunk 2 above)
Comment (glpierce) [reply frontend/src/components/column-selectors/single-column-selector/single-column-selector-list-item.tsx:46]: We need the deep equals to check the name, table of origin, and the relation object if it was joined in on a relation (ie confirm that the "as" and "on" values match within the relation object.

**Pull request merged**